src/gui/qgscoordinatereferencesystemmodel.cpp

```diff
@@ -0,0 +1,751 @@
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#include "qgscoordinatereferencesystemmodel.h"
+#include "qgscoordinatereferencesystemutils.h"
+#include "qgsapplication.h"
+
+#include <QFont>
+
+QgsCoordinateReferenceSystemModel::QgsCoordinateReferenceSystemModel( QObject *parent )
+  : QAbstractItemModel( parent )
+  , mRootNode( std::make_unique< QgsCoordinateReferenceSystemModelGroupNode >( QString(), QIcon(), QString() ) )
+{
+  mCrsDbRecords = QgsApplication::coordinateReferenceSystemRegistry()->crsDbRecords();
+
+  rebuild();
+
+  connect( QgsApplication::coordinateReferenceSystemRegistry(), &QgsCoordinateReferenceSystemRegistry::userCrsAdded, this, &QgsCoordinateReferenceSystemModel::userCrsAdded );
+  connect( QgsApplication::coordinateReferenceSystemRegistry(), &QgsCoordinateReferenceSystemRegistry::userCrsRemoved, this, &QgsCoordinateReferenceSystemModel::userCrsRemoved );
+  connect( QgsApplication::coordinateReferenceSystemRegistry(), &QgsCoordinateReferenceSystemRegistry::userCrsChanged, this, &QgsCoordinateReferenceSystemModel::userCrsChanged );
+}
+
+Qt::ItemFlags QgsCoordinateReferenceSystemModel::flags( const QModelIndex &index ) const
+{
+  if ( !index.isValid() )
+  {
+    return Qt::ItemFlags();
+  }
+
+  QgsCoordinateReferenceSystemModelNode *n = index2node( index );
+  if ( !n )
+    return Qt::ItemFlags();
+
+  switch ( n->nodeType() )
+  {
+    case QgsCoordinateReferenceSystemModelNode::NodeGroup:
+      return index.column() == 0 ? Qt::ItemIsEnabled : Qt::ItemFlags();
+    case QgsCoordinateReferenceSystemModelNode::NodeCrs:
+      return Qt::ItemIsSelectable | Qt::ItemIsEnabled;
+  }
+  BUILTIN_UNREACHABLE
+}
+
+QVariant QgsCoordinateReferenceSystemModel::data( const QModelIndex &index, int role ) const
+{
+  if ( !index.isValid() )
+    return QVariant();
+
+  QgsCoordinateReferenceSystemModelNode *n = index2node( index );
+  if ( !n )
+    return QVariant();
+
+  if ( role == RoleNodeType )
+    return n->nodeType();
+
+  switch ( n->nodeType() )
+  {
+    case QgsCoordinateReferenceSystemModelNode::NodeGroup:
+    {
+      QgsCoordinateReferenceSystemModelGroupNode *groupNode = qgis::down_cast< QgsCoordinateReferenceSystemModelGroupNode * >( n );
+      switch ( role )
+      {
+        case Qt::DecorationRole:
+          switch ( index.column() )
+          {
+            case 0:
+              return groupNode->icon();
+            default:
+              break;
+          }
+          break;
+
+        case Qt::DisplayRole:
+        case Qt::ToolTipRole:
+          switch ( index.column() )
+          {
+            case 0:
+              return groupNode->name();
+
+            default:
+              break;
+          }
+          break;
+
+        case Qt::FontRole:
+        {
+          QFont font;
```

Comment:
Why not going through a QStyledItemDelegate for custom fonts? That way, if a font had been customized via CSS theme, you wouldn't be resetting the font's family/size. 

No strong feelings though.

Reply:
I think that's overkill personally -- the font role is exactly designed for this use case and is considerably simpler vs using a delegate